builder/main.py

```
@@ -515,6 +509,210 @@ def build_fs_image(target, source, env):
+                        with partition.open(fs_path, "w") as dest:
+                            dest.write(item.read_bytes())
+                    except Exception as e:
+                        print(f"Warning: Failed to write file {rel_path}: {e}")
+                        skipped_files.append(str(rel_path))
+
+        # Unmount filesystem
+        base_partition.unmount()
+        
+        # FIX: Clean up FAT tables - remove spurious FF bytes
+        # ESP-IDF expects FAT entries after media descriptor to be 00, not FF
+        print(f"\nCleaning FAT tables...")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)